classes/class-gf-securesubmit.php

```diff
@@ -804,15 +870,57 @@ public function registerInitScripts($form, $field_values, $is_ajax)
             $cc_field = $this->get_hpscredit_card_field($form);
         }
 
+        $use_3DSecure = ($this->getEnable3DSecure() === 'yes' ? true : false);
+
         $args = array(
             'apiKey' => $pubKey,
             'formId' => $form['id'],
             'ccFieldId' => $cc_field['id'],
             'ccPage' => rgar($cc_field, 'pageNumber'),
             'isAjax' => $is_ajax,
             'isSecure' => $cc_field['type'] === 'hpscreditcard',
+            'isCCA' => $use_3DSecure,
+            'isCert' => $this->isCert,
             'baseUrl' => plugins_url('', dirname(__FILE__) . '../'),
         );
+
+        if ($use_3DSecure) {
+            $orderNumber = str_shuffle('abcdefghijklmnopqrstuvwxyz');
+            $data = array(
+                'jti' => str_shuffle('abcdefghijklmnopqrstuvwxyz'),
+                'iat' => time(),
+                'iss' => $this->getEnable3DSecureApiIdentifier(),
+                'OrgUnitId' => $this->getEnable3DSecureOrgUnitId(),
+                'Payload' => array(
+                    'OrderDetails' => array(
+                        'OrderNumber' => $orderNumber,
+                        // Centinel requires amounts in pennies
+                        'Amount' => (100 * 0),
+                        'CurrencyCode' => '840',
+                    ),
+                ),
+            );
+
+            if (!class_exists('HeartlandJWT')) {
+                include_once 'class-heartland-jwt.php';
+            }
+            $jwt = HeartlandJWT::encode($this->getEnable3DSecureApiKey(), $data);
+            $verified = HeartlandJWT::verify($jwt, $this->getEnable3DSecureApiKey());
+
+            $args['ccaData'] = array(
+                'jwt' => $jwt,
+                'orderNumber' => $orderNumber,
+            );
+            file_put_contents(
+                '/tmp/gravity_jwt.log',
+                "DATA:\n" . print_r($data, true). "\n" .
+                "JWT:\n" . $jwt. "\n" .
+                "JWT-Verify:\n" . $verified. "\n" .
+                "ARGS:\n" . print_r($args, true) . "\n",
+                FILE_APPEND
+            );
```

Comment:
@dthawkins :(

Reply:
doh! ill get that removed

Reply:
with that removed, we should be good to go